lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -206,6 +211,100 @@ def setUp(self):
             ]
         )
 
+    @mock.patch('xmodule.x_module.XModuleMixin.has_dynamic_children', mock.Mock(return_value='True'))
+    def test_view_redirect_if_entrance_exam_required(self):
```

Comment:
I've put a fix in place for the score not being updated -- let's see how it works on a new sandbox.  @asadiqbal08, if you can take a closer look at the gating logic that would be great -- I was never able to get around it, even using the direct link, but maybe it has to do with some sort of built-up (or invalid) state on the server.